heft-plugins/heft-webpack4-plugin/src/Webpack4Plugin.ts

```diff
@@ -184,11 +184,31 @@ export default class Webpack4Plugin implements IHeftTaskPlugin<IWebpackPluginOpt
     return this._webpackCompiler;
   }
 
+  private _warnAboutNodeJsIncompatibility(taskSession: IHeftTaskSession): void {
+    const versionMatch: RegExpExecArray | null = /^([0-9]+)\./.exec(process.versions.node); // parse the SemVer MAJOR part
+    if (versionMatch) {
+      const nodejsMajorVersion: number = parseInt(versionMatch[1]);
+      if (nodejsMajorVersion >= 16) {
+        if (!/(^|[^a-z\-])--openssl-legacy-provider($|[^a-z\-])/.test(process.env.NODE_OPTIONS ?? '')) {
```

Comment:
Leave a comment describing what this matches.

Reply:
Fixed